### ISSUE TITLE
r/aws_instance,r/aws_launch_template: Support for AWS Nitro Enclaves

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#15909]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Nitro Enclaves are now generally available.

#### Resources

* [`aws_instance`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance)
* [`aws_launch_template`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template)

#### Data sources

* [`aws_instance`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/instance)
* [`aws_launch_template`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/launch_template)

### Potential Terraform Configuration

```hcl
resource "aws_instance" "example" {
  enclave_options {
    enabled = true
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/10/announcing-general-availability-of-aws-nitro-enclaves/).
[Blog post](https://aws.amazon.com/blogs/aws/aws-nitro-enclaves-isolated-ec2-environments-to-process-confidential-data/).
[User Guide](https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave.html).

Requires [AWS SDK v1.35.17](https://github.com/aws/aws-sdk-go/releases/tag/v1.35.17):

* https://github.com/terraform-providers/terraform-provider-aws/pull/15905

**Comment by hansnielsen** [comment]
I've submitted a PR (#16361) that implements the potential Terraform config above (because that config seems pretty reasonable, and AWS clearly left the door open to adding parameters besides `enabled` in the future).

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.22.0 of the Terraform AWS Provider, later this week. Thank you to @hansnielsen for the implementation. 👍
